alphastats/gui/utils/analysis_helper.py

```diff
@@ -66,44 +113,88 @@ def get_analysis_options_from_dict(method, options_dict):
     submitted = st.button("Submit")
 
     if submitted:
-        return method_dict["function"](**chosen_parameter_dict)
+        with st.spinner("Calculating..."):
+            return method_dict["function"](**chosen_parameter_dict)
 
 
-def between_two_groups(method, options_dict):
+def helper_compare_two_groups(method, options_dict):
     """
-    for Volcano Plot, Differetial Expression Analysis and t-test
+    Helper function to compare two groups for example
+    Volcano Plot, Differetial Expression Analysis and t-test
     selectbox based on selected column
     """
 
     chosen_parameter_dict = {}
     group = st.selectbox(
         "Grouping variable",
-        options=st.session_state.dataset.metadata.columns.to_list(),
+        options=["< None >"] + st.session_state.dataset.metadata.columns.to_list(),
     )
-    if group is not None:
+
+    if group != "< None >":
+
+        col1, col2 = st.columns(2)
 
         unique_values = get_unique_values_from_column(group)
+        
+        with col1:
+
+            group1 = st.selectbox("Group 1", options=unique_values)
+
+        with col2:
+            
+            group2 = st.selectbox("Group 2", options=unique_values)
+
+        chosen_parameter_dict.update(
+            {"column": group, "group1": group1, "group2": group2}
+        )
 
-        group1 = st.selectbox("Group 1", options=unique_values)
+    else:
+
+        col1, col2 = st.columns(2)
+
+        with col1:
```

Comment:
In case you are not aware of this you don't need to use the context `with` but can replace the `st.` with the respective `col`:

```
c1,c2 = st.columns(2)

c1.write('xyz')
```